fsconfig.go

```diff
@@ -118,6 +118,18 @@ type FSConfig interface {
 	// advise using WithDirMount instead. There will be behavior differences
 	// between os.DirFS and WithDirMount, as the latter biases towards what's
 	// expected from WASI implementations.
+	//
+	// # Custom fs.FileInfo
```

Comment:
Here is the integration doc, though obviously our underlying FS will be better when it is out. That said fs.FS may still be easier for some even after that, and so I don't think it is a problem to document this